hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/listeners/PopulateIndexListener.kt

```diff
@@ -32,19 +32,22 @@ class PopulateIndexListener(
       log.error("Failed to process message {}", requestJson, e)
       throw e
     }
-    when (indexRequest.type) {
-      POPULATE_INDEX -> populateIndexService.populateIndex(indexRequest.index!!)
-      POPULATE_PRISONER_PAGE -> populateIndexService.populateIndexWithPrisonerPage(indexRequest.prisonerPage!!)
-      POPULATE_PRISONER -> populateIndexService.populateIndexWithPrisoner(indexRequest.prisonerNumber!!)
-      else -> {
-        "Unknown request type for message $requestJson"
-          .let {
-            log.error(it)
-            throw IllegalArgumentException(it)
-          }
+    try {
+      when (indexRequest.type) {
+        POPULATE_INDEX -> populateIndexService.populateIndex(indexRequest.index!!)
+        POPULATE_PRISONER_PAGE -> populateIndexService.populateIndexWithPrisonerPage(indexRequest.prisonerPage!!)
+        POPULATE_PRISONER -> populateIndexService.populateIndexWithPrisoner(indexRequest.prisonerNumber!!)
+        else -> {
+          "Unknown request type for message $requestJson"
+            .let {
+              log.error(it)
+              throw IllegalArgumentException(it)
+            }
+        }
       }
+    } catch (e: ResponseStatusException) {
+      log.error("Message {} failed with error {}", indexRequest, e.message)
```

Comment:
This feels like the wrong kind of exception for handling messages - it has a constructor with an HTTP status. However I can see how it's convenient for reusing custom exceptions many of which seem to occur for both message / HTTP request processing.

Also, shouldn't this throw something to make the message be retried?